### ISSUE TITLE
Failed building the validation schema for POST: /my-endpoint, due to error can't resolve reference commonSchema# from id #

### ISSUE THREAD

**Issue by edrichhans** [fastify/fastify#3913]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Plugin version

3.0.1

### Node.js version

16.14.2

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.3.1

### Description

Using `$ref` as root reference throws the error `Failed building the validation schema for POST: /my-endpoint, due to error can't resolve reference my-schema# from id #3881 

I tried using it with the plugin and without the plugin, getting the same error.

### Steps to Reproduce

```
fastify.addSchema({
  $id: 'commonSchema',
  type: 'object',
  properties: {
    hello: { type: 'string' }
  }
})

fastify.post('/my-endpoint', {
  handler () {},
  schema: {
    body: { $ref: 'commonSchema#' },
    headers: { $ref: 'commonSchema#' }
  }
})
```

### Expected Behavior

No error. It works in version 3.11

**Comment by climba03003** [comment]
I cannot reproduce your error using the below code.
Can you provide a code that I can run and reproduce the error you face.
```js
import Fastify from "fastify";

const fastify = Fastify();

fastify.addSchema({
  $id: "commonSchema",
  type: "object",
  properties: {
    hello: { type: "string" },
  },
});

fastify.post("/my-endpoint", {
  handler() {},
  schema: {
    body: { $ref: "commonSchema#" },
    headers: { $ref: "commonSchema#" },
  },
});

await fastify.ready();
```

**Comment by climba03003** [comment]
> Is this an intended behavior?

Yes, because `common-schema` is not added to your `ajv` instance.
